code/modules/xenoarcheaology/tools/tools.dm

```diff
@@ -310,3 +310,170 @@
 		usr << browse(null, "window=locater")
 
 	updateSelfDialog()
+
+/obj/item/device/xenoarch_multi_tool
+	name = "xenoarcheology multitool"
+	desc = "Has the features of the Alden-Saraspova counter, a measuring tape, and a depth analysis scanner all in one!"
+	icon_state = "ano_scanner2"
+	item_state = "lampgreen"
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	origin_tech = list(TECH_MAGNET = 3, TECH_ENGINEERING = 3, TECH_BLUESPACE = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 10000,"glass" = 5000)
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_BELT
+	//Multitool specific
+	var/scanning = 1 //Lets start on scanning mode.
+	var/measuring = 0
+
+	//Counter Specific
+	var/last_scan_time = 0
+	var/scan_delay = 25
+
+	//Depth Specific
+	var/list/positive_locations = list()
+	var/datum/depth_scan/current
+
+/obj/item/device/xenoarch_multi_tool/attack_self(var/mob/living/user)
+	var/choice
+	if(!scanning) //Are they not in the view records mode?
+		choice = input(user, "What do you wish to do?") as null|anything in list("Scan/View Records", "Measure", "Detect")
+
+	if(choice == "Scan/View Records" || scanning)
+		if(!scanning)
+			to_chat(usr,"You switch the multitool to scanning mode. Use again to view scan records.")
+			scanning = 1
+			measuring = 0 //Just to make sure!
+			return
+		else
+			var/dat = "<b>Coordinates with positive matches</b><br>"
+			dat += "<A href='?src=\ref[src];clear=0'>== Clear all ==</a><br>"
+			if(current)
+				dat += "Time: [current.time]<br>"
+				dat += "Coords: [current.coords]<br>"
+				dat += "Anomaly depth: [current.depth] cm<br>"
+				dat += "Anomaly size: [current.clearance] cm<br>"
+				dat += "Dissonance spread: [current.dissonance_spread]<br>"
+				var/index = responsive_carriers.Find(current.material)
+				if(index > 0 && index <= finds_as_strings.len)
+					dat += "Anomaly material: [finds_as_strings[index]]<br>"
+				else
+					dat += "Anomaly material: Unknown<br>"
+				dat += "<A href='?src=\ref[src];clear=[current.record_index]'>clear entry</a><br>"
+			else
+				dat += "Select an entry from the list<br>"
+				dat += "<br><br><br><br>"
+			dat += "<hr>"
+			if(positive_locations.len)
+				for(var/index = 1 to positive_locations.len)
+					var/datum/depth_scan/D = positive_locations[index]
+					dat += "<A href='?src=\ref[src];select=[index]'>[D.time], coords: [D.coords]</a><br>"
+			else
+				dat += "No entries recorded."
+			dat += "<hr>"
+			dat += "<A href='?src=\ref[src];refresh=1'>Refresh</a><br>"
+			dat += "<A href='?src=\ref[src];change_mode=1'>Change Multitool Mode</a><br>"
+			dat += "<A href='?src=\ref[src];close=1'>Close</a><br>"
+			user << browse(dat,"window=xenoarch_multi_tool")
+
+	else if(choice == "Measure")
+		measuring = 1
+		to_chat(usr,"You switch the multitool to measure mode.")
+		scanning = 0
+
+	else if(choice == "Detect")
```

Comment:
Likewise, the AS already does this, so you'd just have to tell it to find one, and then give the report to the user.